packages/signature-v4/src/credentialDerivation.spec.ts

```diff
@@ -46,27 +46,27 @@ describe("getSigningKey", () => {
   });
 
   describe("caching", () => {
-    it("should return the same promise when called with the same date, region, service, and credentials", () => {
-      const promise1 = getSigningKey(Sha256, credentials, shortDate, region, service);
-      const promise2 = getSigningKey(Sha256, credentials, shortDate, region, service);
+    it("should return the same promise when called with the same date, region, service, and credentials", async () => {
+      const promise1 = await getSigningKey(Sha256, credentials, shortDate, region, service);
+      const promise2 = await getSigningKey(Sha256, credentials, shortDate, region, service);
       expect(promise1).toBe(promise2);
```

Comment:
The promises can still be compared by using [jest.toStrictEqual](https://jestjs.io/docs/en/expect#tostrictequalvalue)
```suggestion
    it("should return the same promise when called with the same date, region, service, and credentials", () => {
      const promise1 = getSigningKey(Sha256, credentials, shortDate, region, service);
      const promise2 = getSigningKey(Sha256, credentials, shortDate, region, service);
      expect(promise1).toStrictEqual(promise2);
```

The test needs to be updated if we're testing resolved values instead of promises.

Reply:
I probably should have update the test description. After the change, they are not the same promise after the change. I mentioned the potential performance implication in the PR description.